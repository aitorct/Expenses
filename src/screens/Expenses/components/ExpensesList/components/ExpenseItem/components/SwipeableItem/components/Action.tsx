import React from 'react';
import {Animated, StyleSheet, Text} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '@theme/colors';
import fontSizes from '@theme/fontSizes';
import sizes from '@theme/sizes';

const Action = ({
  title,
  icon,
  progress,
}: {
  title: string;
  icon: string;
  progress: Animated.AnimatedInterpolation;
}) => {
  const textOpacity = progress.interpolate({
    inputRange: [0, 0.75, 1],
    outputRange: [0, 0, 1],
  });

  return (
    <Animated.View
      style={[
        styles.container,
        {
          transform: [
            {
              scale: progress,
            },
          ],
        },
      ]}>
      <Icon name={icon} color={colors.icon} size={sizes.largeIcon} />
      <Animated.View
        style={[
          styles.container,
          {
            opacity: textOpacity,
            transform: [
              {
                scale: progress,
              },
            ],
          },
        ]}>
        <Text style={styles.text}>{title}</Text>
      </Animated.View>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: sizes.square,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: fontSizes.actionText,
    color: colors.text,
    paddingTop: sizes.s4,
  },
});

export default Action;
