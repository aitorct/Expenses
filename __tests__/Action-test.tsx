import 'react-native';
import {Animated} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Action from '@screens/Expenses/components/ExpensesList/components/ExpenseItem/components/SwipeableItem/components/Action';

it('Action with no animation started renders correctly', () => {
  const animated = new Animated.Value(0);

  const tree = renderer
    .create(<Action title={'Title'} icon={'eye'} progress={animated} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Action with animation finished renders correctly', () => {
  const animated = new Animated.Value(1);

  const tree = renderer
    .create(<Action title={'Title'} icon={'eye'} progress={animated} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
