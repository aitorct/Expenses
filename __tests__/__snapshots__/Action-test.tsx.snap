// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action with animation finished renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
      "width": 100,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 48,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰈈
  </Text>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
          "paddingTop": 4,
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;

exports[`Action with no animation started renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "transform": Array [
        Object {
          "scale": 0,
        },
      ],
      "width": 100,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 48,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰈈
  </Text>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 0,
          },
        ],
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
          "paddingTop": 4,
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
